### PR TITLE
Plug: check for AppCenter before creating updates button

### PR DESCRIPTION
**Pull request by danrabbit**
Alternative to #97 that checks for the existence of AppCenter before creating the update button

### CODE REVIEW

**Comment by cassidyjames** [comment]
@danrabbit for some reason I'm still not getting an updates button after reinstalling AppCenter from the repos, rebooting, etc.

File: src/Plug.vala [hunk 1]
```diff
@@ -212,15 +212,19 @@ public class About.Plug : Switchboard.Plug {
...
-                warning (e.message);
-            }
-        });
+        Gtk.Button? update_button = null;
+        var appcenter_info = new GLib.DesktopAppInfo ("io.elementary.appcenter");
+        if (appcenter_info != null) {
+            update_button = new Gtk.Button.with_label (_("Check for Updates"));
+            update_button.clicked.connect (() => {
+                try {
+                    Process.spawn_command_line_async ("io.elementary.appcenter --show-updates");
```
Comment (tintou) [inline src/Plug.vala]:
```suggestion
                    appcenter_info.launch_action ("ShowUpdates", null);
```
That way it looks better :)

File: src/Plug.vala [hunk 2]
```diff
@@ -213,15 +213,11 @@ public class About.Plug : Switchboard.Plug {
...
+        var appcenter_info = new GLib.DesktopAppInfo ("io.elementary.appcenter.desktop");
         if (appcenter_info != null) {
             update_button = new Gtk.Button.with_label (_("Check for Updates"));
             update_button.clicked.connect (() => {
-                try {
-                    Process.spawn_command_line_async ("io.elementary.appcenter --show-updates");
-                } catch (Error e) {
-                    warning (e.message);
-                }
+                appcenter_info.launch_action ("ShowUpdates", new GLib.AppLaunchContext ());
```
Comment (cassidyjames) [inline src/Plug.vala]: TIL that's a thing 🤯️

Comment (danrabbit) [reply src/Plug.vala]: Is that an approve :p

**Pull request merged**